lib/ProductOpener/Ingredients.pm

```diff
@@ -677,7 +685,57 @@ sub normalize_fr_a_de_enumeration {
 
 	my $a = shift;
 
-	return join(",", map { normalize_fr_a_de_b($a, $_)} @_);
+	return join(", ", map { normalize_fr_a_de_b($a, $_)} @_);
+}
+
+# English: oil, olive -> olive oil
+# French: huile, olive -> huile d'olive
+
+sub normalize_a_of_b($$$) {
+
+	my $lc = shift;
+	my $a = shift;
+	my $b = shift;
+
+	$a =~ s/\s+$//;
+	$b =~ s/^\s+//;
+
+	if ($lc eq "en") {
+	
+		return $b . " " . $a;
+	}
+	elsif ($lc eq "es") {
+		return $a . " de " . $b;
+	}
+	elsif ($lc eq "fr") {
+		$b =~ s/^(de |d')//;
+
+		if ($b =~ /^(a|e|i|o|u|y|h)/i) {
+			return $a . " d'" . $b;
+		}
+		else {
+			return $a . " de " . $b;
+		}
+	}
+}
+
+
+# Vegetal oil (palm, sunflower and olive)
+# -> palm vegetal oil, sunflower vegetal oil, olive vegetal oil
+
+sub normalize_enumeration($$$) {
+
+	my $lc = shift;
+	my $type = shift;
+	my $enumeration = shift;
+	
+	$log->debug("normalize_enumeration", { type => $type, enumeration => $enumeration }) if $log->is_debug();
+	
+	my $separators = '\(|\)|\/| \/ | - |, |,|'  . $Lang{_and_}{$lc};
```

Comment:
Could it be useful to use `$cbrackets, `$dashes`, `$commas`, or `$stops` here? We already have a `$separators` declared at https://github.com/openfoodfacts/openfoodfacts-server/blob/8732b1f90de8bb21ef65d279fb162f88f8845cfc/lib/ProductOpener/Ingredients.pm#L94, which is hidden by this local variable.

Reply:
I changed to use the $dashes etc.